pyprof/nvtx/nvmarker.py

```diff
@@ -267,22 +288,76 @@ def modMarker(mod, fn_name, args):
 
 
 def add_wrapper(mod, fn_name):
-    assert isfunc(mod, fn_name)
+    # assert isfunc(mod, fn_name)
+
+    config = Config.getInstance()
 
     # Get a pointer to the original function
     func = getattr(mod, fn_name)
 
     # Check if the mod has a string representation
     # and is not a Script or Traced module (used by JIT)
+    # yapf: disable
     s = hasattr(mod, "extra_repr") and (type(mod) is not torch.jit.ScriptModule
                                        ) and (type(mod) is not torch.jit.TopLevelTracedModule)
+    # yapf: enable
 
     def wrapper_func(*args, **kwargs):
         global wrappers_enabled
-
-        if (wrappers_enabled):
+        global call_id
+        global op_to_out_tensor_map
+        global patch_list
+        global call_id_to_op_map
+
+        patch_list.append(call_id)
```

Comment:
definitely think all of this should be extracted out into a different function for cleanliness and readability. Once that is done you could probably get away from globals?

Also, I don't see any of this gated by the new config option?

Reply:
Fixed